src/util/JitterInjection.h

```diff
@@ -0,0 +1,158 @@
+#pragma once
+
+// Copyright 2025 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+/**
+ * Jitter Injection Framework for Finding Concurrency Bugs
+ *
+ * This framework implements probabilistic thread-safe delay injection to
+ * discover threading bugs that are sensitive to specific interleavings. By
+ * introducing random delays at strategic points, we can force different thread
+ * scheduling patterns and expose race conditions, deadlocks, and ordering
+ * violations.
+ *
+ * Design:
+ * - Uses the same RNG seed as stellar-core's testing infrastructure
+ * - Probabilistic: delays only happen with a configured probability
+ * - Easy integration: simple macros that compile away in production
+ * - Reproducible: same seed = same failure (great for debugging)
+ *
+ * Usage:
+ *
+ * 1. Basic delay injection (must be called from main thread or in test
+ * context): JITTER_INJECT_DELAY();  // 10% chance of small delay
+ *
+ * 2. Custom probability:
+ *    JITTER_INJECT_DELAY_PROBABILITY(0.5);  // 50% chance
+ *
+ * 3. Custom delay range (in milliseconds):
+ *    JITTER_INJECT_DELAY_CUSTOM(0.5, 10, 100);  // 50% chance, 10-100ms
+ *
+ * 4. Force context switch without delay:
+ *    JITTER_YIELD();  // Always yields, no delay
+ *
+ * 5. At synchronization points:
+ *    {
+ *        std::lock_guard<std::mutex> lock(myMutex);
+ *        JITTER_INJECT_DELAY();  // After lock acquired
+ *        // critical section
+ *    }
+ *    JITTER_INJECT_DELAY();  // After lock released
+ *
+ * 6. Around condition variable operations:
+ *    myCondVar.wait(lock);
+ *    JITTER_INJECT_DELAY();  // After waking up
+ */
+
+#include "util/Math.h"
+#include <chrono>
+#include <mutex>
+#include <optional>
+#include <random>
+#include <thread>
+
+#ifdef BUILD_THREAD_JITTER
+
+namespace stellar
+{
+
+static thread_local stellar_default_random_engine gJitterRandEngine;
+static uint32_t gJitterRandEngineSeed{0};
+
+class JitterInjector
+{
+  private:
+    // Use the same RNG engine type as stellar-core
+    // Seeded from the test's global RNG seed for reproducibility
+    static std::atomic<uint64_t> sInjectionCount;
+    static std::atomic<uint64_t> sDelayCount;
+
+  public:
+    struct Config
+    {
+        // Default probability % (0 - 100) for delay injection
+        // Default 10% chance of delay at each injection point
+        int32_t defaultProbability{10};
+
+        // Default delay range in nanoseconds
+        uint32_t minDelayNs{100000};
+        uint32_t maxDelayNs{10000000}; // 100 microseconds - 10 milliseconds
```

Comment:
in some sense 10ms is a lot but it's also certainly not the longest a computer can have a hiccup. I also notice you're using int32 for a nanosecond count which means the true max for this subsystem is closer to 4s. I think it might be worth switching to a larger max and also maybe not using a uniform probability across the range of jitters? or even switching to int64 and allowing an even larger max? idk. it seems to me some computers will stall for seconds, or even tens or hundreds of seconds sometimes (eg. when paging to disk or suffering some other massive pathology). you don't want to hit that _super_ often but it might be worth having occasionally possible.